aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,482 @@
+
+It's often convenient to use an ``enum`` for a list. The ``CountryEnumViewModel`` uses the ``CountryEnum`` to generate a list of countries:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryEnumViewModel.cs
+  :language: c#
+  :lines: 3-6
+  :dedent: 3
+  
+.. literalinclude::  forms/sample/final/ViewModels/CountryEnum.cs
+  :language: c#
+  :lines: 1-4,6,8-
+
+The `GetEnumSelectList <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/IHtmlHelper/index.html>`__ method generates a `SelectList <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/SelectList/index.html>`__ object for an enum.
+
+.. literalinclude::   forms/sample/final/Views/Home/IndexEnum.cshtml
+  :language: HTML
+  :emphasize-lines: 4-6
+
+You can decorate your model with the ``Display`` attribute to get a richer UI:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryEnum.cs
+  :language: c#
+  :linenos:
+  :emphasize-lines: 5,7
+
+.. image:: forms/_static/enum.png 
+
+Option Group
+^^^^^^^^^^^^^^^^^^^^^
+
+The HTML  `<optgroup> <https://www.w3.org/wiki/HTML/Elements/optgroup>`__ element is generated when the view model contains one or more `SelectListGroup  <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/SelectListGroup/index.html>`__ objects.
+
+The ``CountryViewModelGroup`` groups the  ``SelectListItem`` elements into the "North America" and "Europe" groups:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryViewModelGroup.cs
+  :language: c#
+  :lines: 6-56
+  :dedent: 3
+  :emphasize-lines: 8-9,17,23,35,41,47,29
+
+The two groups are shown below:
+
+.. image:: forms/_static/grp.png  
+
+Multiple select
+^^^^^^^^^^^^^^^^^^^^^
+
+The Select Tag Helper  will automatically generate the `multiple = "multiple" <https://www.w3.org/TR/html-markup/select.html#select.attrs.multiple>`__  attribute if the property specified in the ``asp-for`` attribute is an ``IEnumerable``. For example, given the following model:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryViewModelIEnumerable.cs
+  :language: c#
+  :emphasize-lines: 6
+
+With the following view:
+
+.. literalinclude::   forms/sample/final/Views/Home/IndexMultiSelect.cshtml
+  :language: HTML
+  :emphasize-lines: 4
+  
+Generates the following HTML:
+
+.. code-block:: HTML  
+  :linenos:
+  :emphasize-lines: 3
+
+  <form method="post" action="/Home/IndexMultiSelect">
+      <select id="CountryCodes" 
+      multiple="multiple" 
+      name="CountryCodes"><option value="MX">Mexico</option>
+  <option value="CA">Canada</option>
+  <option value="US">USA</option>
+  <option value="FR">France</option>
+  <option value="ES">Spain</option>
+  <option value="DE">Germany</option>
+  </select> 
+      <br /><button type="submit">Register</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+No selection
+---------------------
```

Comment:
This should also be a level 2 subsection. Should not show up below "The Select Tag Helper" section in the TOC.